tasks/release.py

```diff
@@ -78,13 +84,163 @@
+
+    # Used dynamically
+    from tasks.modules import GoModule  # noqa
+
+    # Extract the DEFAULT_MODULES dict
+    match = re.search(r"DEFAULT_MODULES = ({.*?})", contents, re.DOTALL | re.MULTILINE)
+
+    assert match, 'Cannot extract "DEFAULT_MODULES" from tasks/modules.py'
+
+    modules = eval(match.group(1))
+
+    return modules
+
+
+def get_default_modules(ctx):
+    """Return DEFAULT_MODULES either from the current file if current agent context or from the agent6 branch.
+
+    Will read the file of the current branch and parse it if agent6 context.
     """
-    List all PR labeled "major_changed" for this release.
+
+    if is_agent6_context:
+        # Read default modules directly from the file
+        with open("tasks/modules.py") as f:
+            modules_file = f.read()
+
+        return read_default_modules(modules_file)
+    else:
+        from tasks.modules import DEFAULT_MODULES
+
+        return DEFAULT_MODULES
+
+
+def set_agent6_context(
+    ctx, version: str | None = None, major_version: int | None = None, allow_stash=False, echo_switch_info=True
+) -> dict:
+    """Change the context to the agent6 branch.
+
+    The branch can be overriden by setting the AGENT6_BRANCH environment variable.
+
+    Note:
+        Should be used only for operations modifying agent6 files.
+        Prefer using agent_context for read only operations.
+
+    Returns:
+        Dict of {'branch_switched', 'stashed', 'base_branch', 'branch'}
+    """
+
+    global is_agent6_context
+
+    base_branch = get_current_branch(ctx)
+
+    if version is None and major_version is None:
+        version = DEFAULT_AGENT6_VERSION
+        branch = DEFAULT_AGENT6_BRANCH
+    elif major_version is not None:
+        version = version or DEFAULT_AGENT6_VERSION
+        branch = DEFAULT_AGENT6_BRANCH
+    else:
+        branch = version[:4] + '.x'
+
+    branch = os.getenv('AGENT6_BRANCH', branch)
+
+    check_version(version, allow_agent6=True)
+    assert version.startswith('6.'), 'Not an agent6 version'
+    # Ensure this branch exists
+    assert len(ctx.run(f'git branch --list {branch}', hide=True).stdout.strip()), f"Branch {branch} does not exist"
+
+    info = {'branch_switched': False, 'stashed': False, 'base_branch': base_branch, 'branch': branch}
+
+    # Already on the target branch
+    if base_branch == branch:
+        return info
+
+    should_stash = check_uncommitted_changes(ctx)
+
+    assert not (should_stash and not allow_stash), "Uncommitted changes detected, aborting"
+
+    # Save + change branch
+    if should_stash:
+        print(color_message("Stashing uncommitted changes", "bold"))
+        print(f'{color_message("Agent6", "bold")}: Stashing uncommitted changes')
+        ctx.run("git stash -u", hide=True)
+        info['stashed'] = True
+    print(f'{color_message("Agent6", "bold")}: Checking out to {branch}')
+    ctx.run(f"git checkout {branch}", hide=True)
```

Comment:
- nit: you should use `switch` now
- Besides, I'm concerned by this change towards a very different branch. If there is a module you haven't imported yet, I assume there is a risk of incompatibility, correct?
How do you ensure this context switching is harmless?
I suppose a safer approach would be to clone the `6.53.x` branch of the repo in another folder and do operations with a new working directory. You might be able to use some tricks to have a light clone using options like `--filter=blob:none`, `--sparse`, `--no-checkout` and/or `--single-branch` (but be careful for instance `reno` needs the full history